fiftyone/server/files/operations.py

```
@@ -0,0 +1,153 @@
+    # call to avoid TOCTOU races between concurrent uploads.  Each chunk
+    # write is also offloaded so the event loop stays free.
+    resolved_path = normalized
+    try:
+        resolved_path, fh = await anyio.to_thread.run_sync(
+            lambda: _sync_resolve_and_open(normalized)
+        )
+        try:
+            async for chunk in stream:
+                await anyio.to_thread.run_sync(lambda c=chunk: fh.write(c))
+        finally:
+            await anyio.to_thread.run_sync(lambda: fh.close())
```

Ruff (0.16.1)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)